Task 3 (Adding routing +about & error pages)

StefanAtanasov98 opened the pull request:
Implementing routing in the app includes:

- Using react-router-dom;
- Adding about page with info about the app;
- Adding error page to show when a path is invalid;
- Adding navigation buttons to go from home to about and vice versa;

![image](https://user-images.githubusercontent.com/103125976/195374748-3c20d6c1-102d-4e35-9f3e-afb448eb26d7.png)

StefanAtanasov98 committed:
Added routing, about & error pages, navigation buttons for home & about page

virzhinia-petkova-redshelf commented on src/components/AboutPage/AboutPage.js:
```diff
@@ -0,0 +1,22 @@
+  const aboutHeading = <h1>About us</h1>;
+  const aboutTxt = 'We have implemented this app to give access to people from all across the globe to the current and also future weather forecast for cities in any country. With the help of few clicks you can find information about the temperature, humidity, pressure, visibility, wind, as well as other weather characteristics for your chosen city of interest.';
+
+  const navigateToHome = () => navigate('/');
```
do you think having the links in your header would be a better idea so you can navigate from each page to each other page

StefanAtanasov98 replied:
I thought about that and what made me not do it was the fact that for example on my home page I don't need a "Home" button and on my about page I don't need an "About" button. I guess I can still have them in my header but I thought this way of doing it is okay - the "Home Page" button is in a central spot on the about page so that it can't be missed.

virzhinia-petkova-redshelf replied:
cool, but what about when you have more pages - are you going to add the links to all pages?

StefanAtanasov98 replied:
Yeah, you're right, I didn't think that far ahead :D So in this case when I'm on the about page and I click the "About" button (which will stay in the header),there should be no reloading, basically no action happening right?

virzhinia-petkova-redshelf commented on src/components/AboutPage/AboutPage.js:
```diff
@@ -0,0 +1,22 @@
+  const navigateToHome = () => navigate('/');
+
+  return (
+    <>
```
here is a good example where a div instead of a fragment would probably be a better fit since you're wrapping an entire page

StefanAtanasov98 committed:
Refactored html and css code

StefanAtanasov98 committed:
Styling and navigation

Added a logo next to 'Weather Forecast', added navigation buttons in header, removed 'Home Page' button from about page

virzhinia-petkova-redshelf commented on src/App.jsx:
```diff
@@ -1,18 +1,32 @@
-      <Header />
-      <HomePage />
-      <Footer />
+      {/* <Header />
```
delete comments if you don't need them

virzhinia-petkova-redshelf commented on src/components/Homepage/Nav/Nav.js:
```diff
@@ -1,16 +1,22 @@
 import { DropdownMenu } from './DropdownMenu.js';
 
 function Nav(props) {
+  // const navigate = useNavigate();
```
delete what you don't need

virzhinia-petkova-redshelf reviewed (approved):
you can address the comments in your next PR
